python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -788,7 +768,8 @@ def _flush(self, _objs=None):
             for obj_id in obj_ids:
                 self.unlock_transaction(obj_id)
 
-            self._lock.release()
+            for obj in objs:
+                obj._releaseParentLock()
```

Comment:
I think this is no longer needed as it should be called by the `with` statement on line 730?